### PR TITLE
Added example scan-hooks

### PR DESCRIPTION
**Pull request by psiinon**
Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
Thanks!

File: scan-hooks/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Scan Hooks run the the packaged Docker scans and provide methods that are called at all stages of the scans.
```
Comment (thc202) [inline scan-hooks/README.md:1]: `Scan Hooks are run by the packaged...`?

File: scan-hooks/export_session.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# This scan-hook (https://www.zaproxy.org/docs/docker/scan-hooks/)
```
Comment (thc202) [inline scan-hooks/export_session.py:1]: `scan hook`? (To match what's used in the `README`.)

File: scan-hooks/export_session.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+# copies the ZAP session and log file into a directory mounted to /zap/wrk after ZAP has shut down.
+# There must not be a local 'session' directory otherwise the script will fail.
+
+from shutil import copy2, copytree
+import os.path
+
+dev_path = '/home/zap/.ZAP_D'
+rel_path = '/home/zap/.ZAP'
```
Comment (thc202) [inline scan-hooks/export_session.py:9]: Worth mentioning the base path when running as root?

File: scan-hooks/template.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
+# A template scan-hook (https://www.zaproxy.org/docs/docker/scan-hooks/)
```
Comment (thc202) [inline scan-hooks/template.py:1]: Same comment regarding the `scan hook`.

File: scan-hooks/template.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+def cli_opts(opts):
+	print("cli_opts(" + str(opts) + ")")
```
Comment (thc202) [inline scan-hooks/template.py:4]: `print("cli_opts(%s)" % opts)`? (Similar comment for following prints.)

File: scan-hooks/template.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+def zap_started(zap, target):
+	print("zap_started(" + str(zap) + "," + target + ")")
+
+def importing_openapi(target_url, target_file):
```
Comment (thc202) [inline scan-hooks/template.py:9]: Does it worth noting that not all hooks are called in all scans?

**Comment by psiinon** [comment]
Have fixed all of the issues apart from the print statements, they'll take a bit more work and testing :/

File: scan-hooks/export_session.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+# This scan hook (https://www.zaproxy.org/docs/docker/scan-hooks/)
+# copies the ZAP session and log file into a directory mounted to /zap/wrk after ZAP has shut down.
+# There must not be a local 'session' directory otherwise the script will fail.
+
+from shutil import copy2, copytree
+import os.path
+
+# This assumes that you are running as zap - if you run as root change to use /home/root/
```
Comment (thc202) [inline scan-hooks/export_session.py:8]: The path needs to be tweaked, should be e.g. `/root/.ZAP`.

**Comment by psiinon** [comment]
Updated based on the comments :)

**Comment by thc202** [comment]
Thanks!

**Pull request merged**